src/cljs/rems/applications.cljs

```diff
@@ -37,6 +40,32 @@
  (fn [db _]
    (::loading-my-applications? db)))
 
+;;;; all applications
+
+(rf/reg-event-fx
+ ::fetch-all-applications
+ (fn [{:keys [db]} _]
+   (fetch "/api/applications"
+          {:handler #(rf/dispatch [::fetch-all-applications-result %])})
+   {:db (assoc db ::loading-all-applications? true)}))
+
+(rf/reg-event-db
+ ::fetch-all-applications-result
+ (fn [db [_ applications]]
+   (-> db
+       (assoc ::all-applications applications)
+       (dissoc ::loading-all-applications?))))
+
+(rf/reg-sub
+ ::all-applications
+ (fn [db _]
+   (::all-applications db)))
+
+(rf/reg-sub
+ ::loading-all-applications?
+ (fn [db _]
+   (::loading-all-applications? db)))
+
```

Comment:
These are very similar to my-applications. Would it make sense to make a generic that takes an URL and a db path, and stores the data and loading status under a map in that path?

Or is that abusing re-frame or just premature abstraction, as this is just the second version and only about 20 lines?

Reply:
Similar loading is done on also other pages, so a generic component for doing that would make sense.